onnxruntime/python/backend/backend.py

```diff
@@ -107,7 +107,16 @@ def prepare(cls, model, device=None, **kwargs):
             for k, v in kwargs.items():
                 if hasattr(options, k):
                     setattr(options, k, v)
-            inf = InferenceSession(model, sess_options=options, providers=get_available_providers())
+
+            providers = get_available_providers()
+
+            if os.getenv('ORT_ONNX_BACKEND_EXCLUDE_PROVIDERS'):
```

Comment:
would be more compact to write it like this:

excluded_providers = os.getenv('ORT_ONNX_BACKEND_EXCLUDE_PROVIDERS', default="").split(',')
providers = [x for x in providers if (x not in excluded_providers)]